### DISCUSSION TITLE
No serverless pages were built

### DISCUSSION THREAD

**Discussion by adelinger** [vercel/community#1212] [Help] [answered]
I am trying to deploy my new Nextjs project to Vercel, but I am getting no serverless pages were built error. 

I also get this link: Learn More: https://err.sh/vercel/vercel/now-next-no-serverless-pages-built
But as I can see, this answer is outdated and does not have anything to do with my project. 

Does anyone have any other suggestions as to what should I look for or why this could happen? 

I tried with simplest next.config.js possible and still it's not working, and I tried the same settings I had on other project - also same error. 

Any help is appriciated. Thanks.

Here is link on one of my recent deployments:
https://vercel.com/adelinger/at-shop/ALYFGZC1Q7wVXPAuaZdVfK6EvxQK

P.S. I googled for hours and could not find aynthing (except outdated answers).

![Screenshot 2023-01-05 at 20 27 50](https://user-images.githubusercontent.com/31591846/210864558-ebf69a1c-cf60-417e-b028-dd6227550f19.png)

**Comment by amyegan** [comment]
Thanks for reaching out @adelinger! Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or public repo that I can use to get a closer look at the problem?

**Comment by adelinger** [comment]
Yes, take a look here:
https://github.com/adelinger/at-shop/tree/develop

**Reply by amyegan** [reply]
Is this code running for you on localhost if you use `npm run dev`? I'm not able to get a development or preview build to work on my computer 😖

**Comment by adelinger** [comment, marked as answer]
Yes, sure thing, npm run dev, start or build. Everything works. 

Develop branch, right?

**Reply by amyegan** [reply]
Yes, the `develop` branch is what I used. I suspect you have something installed either in the local copy of your repo or on your local environment that is allowing the build to work for you, but somehow that dependency did not make it into your `package.json`.  It looks like Tailwind CSS is one of them. I see you have some references to `tailwindcss`, but I don't see it in your dependencies so it isn't being installed. I tried adding it, which solved one problem, but the overall build still isn't working for me so there are probably some other missing dependencies.

**Reply by adelinger** [reply]
Hi. I just tried running my project on two other machines and yes, it was as you said, I had problems with the dependencies. 
But I fixed those now and I could run npm run dev on both devices. So now I have fully functioning project running on three machines. 

Unfortunately deploy on Vercel is still not working, giving me the same error. Could you please check my project now? 
Thanks!

**Reply by amyegan** [reply]
I just tried again, but still no luck. 🙁 

On the latest `develop` branch, I get a blank page from `npm run dev`. And if I try to build or deploy, the build step never seems to finish. It eventually times out with the `No serverless pages were built` error message.

**Reply by adelinger** [reply]
Hm, I had the same problem on other two machines (blank page in browser on npm run dev). Npm run start took too long so I did not wait for it to finish, but I assumed that PC is just slow or something... Those two machines are Windows PC's. But on my main computer it is all working fine (Mac OS) and I developed whole project on it, everything always worked until I tried to deploy it on Vercel. 

Any ideas why is it happening? I have no clue.

**Reply by amyegan** [reply]
I'm guessing there is another missing dependency or environment variable that you have on your Mac but not on your other computer. I'm also running on Mac OS and getting the same result you describe for your Windows PC, which sounds like a missing dependency and/or uncaught exception.

**Reply by adelinger** [reply]
Yes, but should not be enviroment variables since I added those when deploying on vercel. 
I am lost now beucase I don't know where to look. How to find missing dependencies? I don't get any error so have no idea where to look at.

**Reply by adelinger** [reply]
> I'm guessing there is another missing dependency or environment variable that you have on your Mac but not on your other computer. I'm also running on Mac OS and getting the same result you describe for your Windows PC, which sounds like a missing dependency and/or uncaught exception.

I think I am getting somewhere... I finally resolved dependencies mess, now npm run dev works on all my machines, also npm run build and npm run start. But my deployed app gives me 500 error. 
What could that be?

**Reply by adelinger** [reply]
Apparently, problem is about plaiceholder and blur.png
![Screenshot 2023-01-10 at 18 49 57](https://user-images.githubusercontent.com/31591846/211625349-05d6be1c-212e-4f5f-96cd-0d45217b5750.png)

**Reply by amyegan** [reply]
Looks like the image file isn't being pulled into the `getServerSideProps` function for some reason. 🤔 

You could work around this by using the deployed address of the image, like
```
const blurImageAddress = "https://" + process.env.VERCEL_URL + blurImageUrl;
const { base64, img } = await getPlaiceholder(blurImageAddress);
```

Or you may be able to force the image path into the function by adding 
```
path.join(process.cwd(), "public", blurImageUrl);
```

**Reply by adelinger** [reply]
Ok so that's it, now everything works fine. It was a problem with dependencies and later on I fixed this problem with blur image like you said. 
Thank you very much, this was very helpful.

**Comment by Chukwu3meka** [comment]
Navigate to settings of your project in Vercel dashboard  

> https://vercel.com/[username]/[project]/settings

### Follow this process to fix:

1. Scroll down to **Node.js Version** of Search for **Node.js Version** on the page
2. Select a newer version, probably **16x** or **18x** and not 14x 
3. Click the 'SAVE' button
4. Navigate back to **Deployments** https://vercel.com/[username]/[project]/deployments
5. Click the hamburger menu (three dots) in your most recent commit
6. Then click the 'REDEPLOY' button and make sure not to select the cache option

App should build successfully now

**Comment by Aymenjdily** [comment]
I am still facing the same issue, i am always deploying my projects without any issue like that, is it something wrong with vercel ? because npm run build work well with me, but when deploying i got "Error: No serverless pages were built"

**Reply by Aymenjdily** [reply]
@amyegan

**Reply by amyegan** [reply]
If the Node.js version selected in Project Settings is 16 or lower, then let's try with a more recent version. Select `18.x` or `20.x` and save the change, go back to the deployments list and re-deploy your project, and let me know if that worked for you 🙏

**Reply by Aymenjdily** [reply]
in my project am using node 20. and the same in vercel but still having the same problem

**Reply by amyegan** [reply]
Thanks for letting me know. I see you also shared more details in https://github.com/vercel/community/discussions/5109. Let's continue the conversation there

**Comment by Aymenjdily** [comment]
in my project am using node 20. and the same in vercel but still having
the same problem
